modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/new-js/FormBuilder.js

```diff
@@ -51,6 +58,50 @@ export const FormBuilder = () => {
 		[initialSidebarPanels, portletNamespace]
 	);
 
+	useEffect(() => {
+		if (allowFieldSets && contentType) {
+			let globalFieldSetsPromise = [];
+
+			if (groupId) {
+				globalFieldSetsPromise = getItem(
+					`/o/data-engine/v2.0/sites/${groupId}/data-definitions/by-content-type/${contentType}`
+				);
+			}
+
+			const groupFieldSetsPromise = getItem(
+				`/o/data-engine/v2.0/data-definitions/by-content-type/${contentType}`
+			);
+
+			const fetchFieldSets = async () => {
+				try {
+					const [
+						{items: globalFieldSets = []},
+						{items: groupFieldSets = []},
+					] = await Promise.all([
+						globalFieldSetsPromise,
+						groupFieldSetsPromise,
+					]);
+
+					dispatch({
+						payload: {
+							fieldSets: [...globalFieldSets, ...groupFieldSets],
+						},
+						type: SIDEBAR.FIELDSET.UPDATE,
```

Comment:
It would be good to avoid calling directly, the `EVENT_TYPES` contract is `EVENT_TYPES.SIDEBAR.FIELDSET.UPDATE`.